src/include/homestore/blk.h

```diff
@@ -141,6 +141,30 @@ std::basic_ostream< charT, traits >& operator<<(std::basic_ostream< charT, trait
 
     return outStream;
 }
+
+/* Hints for various allocators */
+struct blk_alloc_hints {
+    blk_alloc_hints() :
+            desired_temp{0},
+            dev_id_hint{INVALID_DEV_ID},
+            can_look_for_other_chunk{true},
+            is_contiguous{false},
+            multiplier{1},
+            max_blks_per_entry{BlkId::max_blks_in_op()},
+            stream_info{(uintptr_t) nullptr} {}
+
+    blk_temp_t desired_temp;       // Temperature hint for the device
+    uint32_t dev_id_hint;          // which physical device to pick (hint if any) -1 for don't care
+    bool can_look_for_other_chunk; // If alloc on device not available can I pick other device
+    bool is_contiguous;
+    uint32_t multiplier;         // blks allocated in a blkid should be a multiple of multiplier
+    uint32_t max_blks_per_entry; // Number of blks on every entry
+    uintptr_t stream_info;
+#ifdef _PRERELEASE
+    bool error_simulate = false; // can error simulate happen
+#endif
```

Comment:
Justa nit pick, we could set default values in the declaration itself instead of calling them in constructor.

Reply:
Curious why we had to move this here from blk_allocator.h?